netty/src/test/java/io/grpc/netty/NettyClientStreamTest.java

```diff
@@ -392,6 +393,38 @@ public void removeUserAgentFromApplicationHeaders() {
         .containsEntry(Utils.USER_AGENT, AsciiString.of("good agent"));
   }
 
+  @Test
+  public void getRequestSentThroughHeader() {
```

Comment:
A test like this should be in AbstractClientStream2Test.java as well. At that point, this one could possibly become smaller.